cmd/soroban-cli/src/commands/snapshot/create.rs

```diff
@@ -215,29 +218,35 @@ impl Cmd {
             wasm_hashes: self.wasm_hashes.iter().cloned().collect(),
         };
         let mut next = SearchInputs::default();
+
         loop {
             if current.is_empty() {
                 break;
             }
-            println!(
-                "ℹ️  Searching for {} accounts, {} contracts, {} wasms",
+
+            print.infoln(format!(
+                "Searching for {} accounts, {} contracts, {} wasms",
                 current.account_ids.len(),
                 current.contract_ids.len(),
                 current.wasm_hashes.len(),
-            );
+            ));
+
             for (i, bucket) in buckets.iter().enumerate() {
                 // Defined where the bucket will be read from, either from cache on
                 // disk, or streamed from the archive.
-                let cache_path = cache_bucket(&archive_url, i, bucket).await?;
+                let cache_path = cache_bucket(&print, &archive_url, i, bucket).await?;
                 let file = std::fs::OpenOptions::new()
                     .read(true)
                     .open(&cache_path)
                     .map_err(Error::ReadOpeningCachedBucket)?;
-                print!("🔎 Searching bucket {i} {bucket}");
+
+                let message = format!("Searching bucket {i} {bucket}");
+                print.search(format!("{message}…"));
+
                 if let Ok(metadata) = file.metadata() {
-                    print!(" ({})", ByteSize(metadata.len()));
+                    print.clear_line();
```

Comment:
Yep, `\r` works everywhere. I just need to check Windows when I have the chance, to ensure that it clears the whole line. Otherwise, there are other hacks we can use.